services/mentor_service.js

```diff
@@ -73,8 +70,18 @@ async function getEligibleMentorStatuses(userRoles, isOwner) {
   }
 }
 
+async function updateMentorStatus(userRoles, mentorId, status, isOwner) {
+  const isAdmin = userRoles.some(role => role.role_name === ROLE_NAMES.admin);
+  if(isAdmin || isOwner) {
+    return repo.updateMentorStatus(mentorId, status)
```

Comment:
Please don't forget about:
* Mentor role is assigned to user on application approval (status change from pending to active)